### PR TITLE
feat: add enable analytics checkbox for signup

### PR DESCRIPTION
**Pull request by ReidyT**
This code is an updated version of #70.

In this version, a checkbox is used instead of a switch to follow the same UI as for the user agreements:

image

### CODE REVIEW

File: cypress/e2e/SignUp.cy.ts [hunk 1]
```diff
@@ -101,4 +102,39 @@ describe('SignUp', () => {
       cy.get(`#${SIGN_UP_BUTTON_ID}`).should('be.visible');
     });
   });
+
+  describe('Defining Analytics On Sign Up', () => {
+    const { GRAASP } = MEMBERS;
+
+    beforeEach(() => {
+      cy.visit(SIGN_UP_PATH);
+      // eslint-disable-next-line arrow-body-style
```
Comment (pyphilia) [inline cypress/e2e/SignUp.cy.ts:111]: Remove?

File: src/langs/en.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
   "SIGN_UP_HEADER": "Sign Up",
   "SIGN_UP_LINK_TEXT": "Not registered? Click here to register",
   "SIGN_UP_SAVE_ACTIONS_LABEL": "Enable Analytics",
-  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "Coming Soon!",
+  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "We collect analytics data to improve your user experience while browsing Graasp.",
```
Comment (pyphilia) [inline src/langs/en.json:24]:
```suggestion
  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "We collect analytic data to improve your user experience while browsing Graasp.",
```

File: src/langs/fr.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
   "SIGN_UP_HEADER": "S'enregistrer",
   "SIGN_UP_LINK_TEXT": "Pas encore enregistré ? Cliquez ici pour vous enregistrer",
   "SIGN_UP_SAVE_ACTIONS_LABEL": "Activer les statistiques d'utilisation",
-  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "Prochainement!",
+  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "Nous recueillons des données analytiques afin d'améliorer votre expérience d'utilisateur lorsque vous naviguez sur Graasp.",
```
Comment (pyphilia) [inline src/langs/fr.json:24]:
```suggestion
  "SIGN_UP_SAVE_ACTIONS_TOOLTIP": "Nous recueillons des données analytiques afin d'améliorer votre expérience utilisateur lorsque vous naviguez sur Graasp.",
```

File: src/hooks/useEnableAnalyticsForm.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+
+export type UseEnableAnalyticsForm = {
+  enableSaveActions: boolean;
+  updateSaveActions: (enable: boolean) => void;
+};
+
+export const useEnableAnalyticsForm = (): UseEnableAnalyticsForm => {
```
Comment (pyphilia) [inline src/hooks/useEnableAnalyticsForm.tsx:8]: I'm not against hooks, but this one look especially short?

**Review by pyphilia — APPROVED** [review]
Thanks for the PR, I would maybe like to have a longer text for `Enable Analytics`, not sure what. From chatgpt: `Activate data-driven insights saving for enhanced analytics dashboards`. You can shorten aha 🚋

File: src/hooks/useEnableAnalyticsForm.tsx (hunk 4 above)
Comment (ReidyT) [reply src/hooks/useEnableAnalyticsForm.tsx:8]: Yes I will set the state in the sign up.

**Pull request merged**